e2e/testcafe-devextreme/tests/dataGrid/markup/T1240074_hoveringRows.ts

```diff
@@ -0,0 +1,27 @@
+import DataGrid from 'devextreme-testcafe-models/dataGrid';
+import { createWidget } from '../../../helpers/createWidget';
+import url from '../../../helpers/getPageUrl';
+import { getData } from '../helpers/generateDataSourceData';
+
+fixture.disablePageReloads`HoveringRows`.page(
+  url(__dirname, '../../container.html'),
+);
+
```

Comment:
Let's extend the existing test case to check more cases
Now we're checking only that the first row has the correct class on hover
But in the original issue, the last row had the correct class on hover (not the first, but I mean that checking only the first row may be an edge case)

So, let's do something like this:
```typescript
test('Hover over rows in the middle', async(t) => {
  const dataGrid = new DataGrid('#container');
  const firstRow = dataGrid.getDataRow(10);
  const secondRow = dataGrid.getDataRow(11);
  
  await t.hover(firstRow.element)
             .expect(firstRow.isHovered)
             .ok();

  await t.hover(secondRow.element)
             .expect(firstRow.isHovered)
             .notOk()
             .expect(secondRow.isHovered)
             .ok();
}).before(async () => {
  await createWidget(
    'dxDataGrid',
    {
      dataSource: getData(20, 3),
      hoverStateEnabled: true,
    }
  )
});

test('Hover over first and last rows', async (t) => {
const dataGrid = new DataGrid('#container');
  const firstRow = dataGrid.getDataRow(10);
  const lastRow = dataGrid.getDataRow(20);
  
  await t.hover(firstRow.element)
             .expect(firstRow.isHovered)
             .ok();

  await t.hover(lastRow.element)
             .expect(firstRow.isHovered)
             .notOk()
             .expect(lastRow.isHovered)
             .ok();
}).before(async () => {
  await createWidget(
    'dxDataGrid',
    {
      dataSource: getData(20, 3),
      hoverStateEnabled: true,
    }
  )
});
```